src/DCSAM.cpp

```diff
@@ -30,8 +30,18 @@ void DCSAM::update(const gtsam::NonlinearFactorGraph &graph,
                    const gtsam::DiscreteFactorGraph &dfg,
                    const DCFactorGraph &dcfg,
                    const gtsam::Values &initialGuessContinuous,
-                   const DiscreteValues &initialGuessDiscrete) {
-  // First things first: combine currContinuous_ estimate with the new values
+                   const DiscreteValues &initialGuessDiscrete,
+                   const gtsam::FactorIndices &removeFactorIndices,
+                   const std::vector<size_t> &removeDiscreteFactorIndices) {
+
+  // First things first: get rid of factors that are to be removed so updates
+  // to follow take the removals into account 
+  isam_.update(gtsam::NonlinearFactorGraph(), gtsam::Values(), removeFactorIndices);
+  for (auto& i : removeDiscreteFactorIndices) {
+    dfg_.remove(i);
+  }
+
+  // Next: combine currContinuous_ estimate with the new values
   // from initialGuessContinuous to produce the full continuous variable state.
```

Comment:
The implementations look good to me. We will need to also track all the factor indices (for both discrete & continuous)